Sources/SwiftLexicalLookup/SimpleLookupQueries.swift

```diff
@@ -0,0 +1,227 @@
+  ///
+  @_spi(Experimental) public func lookupCatchNode() -> Syntax? {
+    return lookupCatchNodeHelper(at: Syntax(self), traversedCatchClause: false)
+  }
+
+  // MARK: - lookupLabeledStmts
+
+  /// Given syntax node position, returns all available labeled statements.
+  private func lookupLabeledStmts(at syntax: SyntaxProtocol) -> [LabeledStmtSyntax] {
+    return walkParentTreeUpToFunctionBoundary(
+      at: syntax.parent,
+      collect: LabeledStmtSyntax.self
+    )
+  }
+
+  // MARK: - lookupCatchNode
+
+  /// Given syntax node location, finds where an error could be caught. If set to `true`, `traverseCatchClause`lookup will skip the next do statement.
+  private func lookupCatchNodeHelper(at syntax: Syntax?, traversedCatchClause: Bool) -> Syntax? {
+    guard let syntax else { return nil }
+
+    switch syntax.as(SyntaxEnum.self) {
+    case .doStmt:
+      if traversedCatchClause {
+        return lookupCatchNodeHelper(at: syntax.parent, traversedCatchClause: false)
+      } else {
+        return syntax
+      }
+    case .catchClause:
+      return lookupCatchNodeHelper(at: syntax.parent, traversedCatchClause: true)
+    case .tryExpr(let tryExpr):
+      if tryExpr.questionOrExclamationMark != nil {
+        return syntax
+      } else {
+        return lookupCatchNodeHelper(at: syntax.parent, traversedCatchClause: traversedCatchClause)
+      }
+    case .functionDecl, .accessorDecl, .initializerDecl, .deinitializerDecl, .closureExpr:
+      return syntax
+    case .exprList(let exprList):
+      if let tryExpr = exprList.first?.as(TryExprSyntax.self), tryExpr.questionOrExclamationMark != nil {
+        return Syntax(tryExpr)
+      }
+      return lookupCatchNodeHelper(at: syntax.parent, traversedCatchClause: traversedCatchClause)
+    default:
+      return lookupCatchNodeHelper(at: syntax.parent, traversedCatchClause: traversedCatchClause)
+    }
+  }
+
+  // MARK: - walkParentTree helper methods
+
+  /// Callect the first syntax node matching the collection type up to a function boundary.
+  fileprivate func walkParentTreeUpToFunctionBoundary<T: SyntaxProtocol>(
+    at syntax: Syntax?,
+    collect: T.Type
+  ) -> T? {
+    walkParentTreeUpToFunctionBoundary(at: syntax, collect: collect, stopWithFirstMatch: true).first
+  }
+
+  /// Callect syntax nodes matching the collection type up to a function boundary.
+  fileprivate func walkParentTreeUpToFunctionBoundary<T: SyntaxProtocol>(
+    at syntax: Syntax?,
+    collect: T.Type,
+    stopWithFirstMatch: Bool = false
+  ) -> [T] {
+    walkParentTree(
+      upTo: [
+        MemberBlockSyntax.self,
+        FunctionDeclSyntax.self,
+        InitializerDeclSyntax.self,
+        DeinitializerDeclSyntax.self,
+        AccessorDeclSyntax.self,
+        ClosureExprSyntax.self,
+      ],
+      collect: collect,
+      stopWithFirstMatch: stopWithFirstMatch
+    )
+  }
+
+  /// Callect syntax nodes matching the collection type up until encountering one of the specified syntax nodes.
```

Comment:
```suggestion
  /// Collect syntax nodes matching the collection type up until encountering one of the specified syntax nodes. The nodes in the array are inside out, with the innermost node being the first.
```

Same in a couple more locations